docs/explanation/e-interfaces.md

```diff
@@ -0,0 +1,50 @@
+# Interfaces/endpoints
+
+Charmed MySQL K8s supports modern `mysql_client` and legacy `mysql` interfaces (in a backward compatible mode).
+
+**Note:** do NOT relate both modern and legacy interfaces simultaneously!
+
+## Modern interfaces
+
+This charm provides modern ['mysql_client' ](https://github.com/canonical/charm-relation-interfaces) interface. Applications can easily connect MySQL using ['data_interfaces'](https://charmhub.io/data-platform-libs/libraries/data_interfaces) library from ['data-platform-libs'](https://github.com/canonical/data-platform-libs/).
```

Comment:
extra space in `['mysql_client' ]`

Reply:
Fixed on Discourse, it will be synced here the next PR.